### PR TITLE
Gtests for AQBase and LSGC

### PR DESCRIPTION
**Pull request by jsrehak**
Adds unit tests for `AQBase` and `LSGC`. Makes various small changes, such as adding `const` to various `AQBase` functions, and overloading the `PrintAQ` function. Addresses parts of #69 and #67.

### CODE REVIEW

File: src/aqdata/aq_base.cc [hunk 1]
```diff
@@ -39,8 +39,8 @@ void AQBase<dim>::InitRefBCInd () {
   // uses deal.II generated mesh or mesh from gmsh with proper
   // boundary IDs setup: {0,1,2,3,4,5} for {xmin,xmax,ymin,ymax,zmin,zmax}
   if (have_reflective_bc_) {
-    Assert (dim>1,
-            dealii::ExcNotImplemented());
+    AssertThrow (dim>1,
```
Comment (weixiong-zheng-berkeley) [inline src/aqdata/aq_base.cc]: Hmm, the reflective boundary is actually implemented for general dimension and should work in 1D, so please delete this but maybe add a `TODO` comment for testing for it.

File: src/aqdata/aq_base.cc [hunk 2]
```diff
@@ -111,74 +111,79 @@ template <int dim>
...
 }
 
 template <int dim>
-std::string AQBase<dim>::ProduceAQName () {
+std::string AQBase<dim>::ProduceAQName () const {
   AssertThrow (aq_name_.size()>0,
                dealii::ExcMessage("aq name has to be assigned"));
   // ToDo: more quadrature name producers
-  if (aq_name_=="lsgc")
+  if (aq_name_== "lsgc")
```
Comment (weixiong-zheng-berkeley) [inline src/aqdata/aq_base.cc]: I suggest if you think a white space is needed on one side of the operator, add another one to the other side, i.e. `aq_name_ == "lsgc"`

File: src/aqdata/tests/aq_base_test.cc [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  std::string filename = "aq_base_1d_ep";
+  prm.set("transport model", "ep");
+  btest::GoldTestInit(filename); // Opens deal log
+  OutputAQ<1>();
+  btest::GoldTestRun(filename); // Closes deal log
+}
+
+TEST_F(AQBaseTest, AQBaseBadDimProduceAQ) {
+  AQBase<2> test_AQ(prm);
+  ASSERT_THROW(test_AQ.ProduceAQ(), dealii::ExcMessage);
```
Comment (weixiong-zheng-berkeley) [inline src/aqdata/tests/aq_base_test.cc]: I've never used `dealii::ExcMessage` without a message and I didn't find a overload of this kind. What does it do?

File: tests/aqdata/CMakeLists.txt [hunk 4]
```diff
@@ -1,25 +0,0 @@
-# Add a new executable for every single new test
```
Comment (weixiong-zheng-berkeley) [inline tests/aqdata/CMakeLists.txt]: Eventually! Great.

File: src/aqdata/tests/aq_base_test.cc (hunk 3 above)
Comment (jsrehak) [reply src/aqdata/tests/aq_base_test.cc]: This is just asserting that `dealii::ExcMessage` is thrown when you try to `AQBase.ProduceAQ` with `dim > 1`.

File: src/aqdata/tests/lsgc_test.cc [hunk 5]
```diff
@@ -14,18 +30,13 @@ class LSGCTest : public ::testing::Test {
...
                        dealii::Patterns::Integer (), "");
-  }
-
-  template <int dim>
-  void AQDataTest();
-  
-  dealii::ParameterHandler prm;
-};
+    prm.declare_entry ("transport model", "regular",
+                     dealii::Patterns::Selection("regular|ep"), "");
```
Comment (weixiong-zheng-berkeley) [inline src/aqdata/tests/lsgc_test.cc]: Either add two more space before `dealii::Patterns...` or put it forward with only four more indent spaces (6 in total), i.e. 
```prm.declare_entry ("transport model", "regular",
    dealii::Patterns::Selection("regular|ep"), "");```

Comment (jsrehak) [reply src/aqdata/tests/lsgc_test.cc]: Good eye! I clearly missed hitting tab there.

File: src/aqdata/aq_base.cc (hunk 1 above)
Comment (jsrehak) [reply src/aqdata/aq_base.cc]: Exception removed and test added.

**Pull request merged**